pkg/plugins/policies/meshtimeout/plugin/xds/configurer.go

```diff
@@ -0,0 +1,118 @@
+package xds
+
+import (
+	"time"
+
+	envoy_cluster "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	envoy_route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	envoy_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	envoy_tcp "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/tcp_proxy/v3"
+	envoy_upstream_http "github.com/envoyproxy/go-control-plane/envoy/extensions/upstreams/http/v3"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/types/known/durationpb"
+	kube_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	api "github.com/kumahq/kuma/pkg/plugins/policies/meshtimeout/api/v1alpha1"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+	clusters_v3 "github.com/kumahq/kuma/pkg/xds/envoy/clusters/v3"
+	listeners_v3 "github.com/kumahq/kuma/pkg/xds/envoy/listeners/v3"
+)
+
+const (
+	defaultConnectionTimeout     = time.Second * 5
+	defaultIdleTimeout           = time.Hour
+	defaultRequestTimeout        = time.Second * 15
+	defaultStreamIdleTimeout     = time.Minute * 30
+	defaultMaxStreamDuration     = 0
+	defaultMaxConnectionDuration = 0
```

Comment:
does 0 mean 0 or disabled? if disabled then should any be disabled? I know that disabling idle timeouts may lead to connection leaks.

Reply:
Both of those timeouts are disabled by default in envoy. We can set some specific timeout. Do you have some value in mind?